### PR TITLE
✨ Add getProcessModelByProcessInstanceId

### PR DESCRIPTION
**Pull request by SteffenKn**
> Related to https://github.com/process-engine/process_engine_runtime/pull/207

**Changes:**

1. Add getProcessModelByProcessInstanceId

PR: #30

## How can others test the changes?

-  **Have a look at https://github.com/process-engine/bpmn-studio/pull/1225**

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Fixing Security Issues   | `:lock:`             | 🔒     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️      |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️      |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: typescript/src/accessors/internal_accessor.ts [hunk 1]
```diff
@@ -127,6 +127,10 @@ export class InternalAccessor implements IConsumerApiAccessor {
     return this._consumerApiService.getProcessModelById(identity, processModelId);
   }
 
+  public async getProcessModelByProcessInstanceId(identity: IIdentity, processInstanceId: string): Promise<DataModels.ProcessModels.ProcessModel> {
+    return this._consumerApiService.getProcessModelByProcessInstanceId(identity, processInstanceId);
```
Comment (robpal94) [inline typescript/src/accessors/internal_accessor.ts]: Isn't there an `await` missing? 

```suggestion
    return await this._consumerApiService.getProcessModelByProcessInstanceId(identity, processInstanceId);
```

File: typescript/src/accessors/internal_accessor.ts [hunk 2]
```diff
@@ -183,6 +187,10 @@ export class InternalAccessor implements IConsumerApiAccessor {
     return this._consumerApiService.getUserTasksForProcessModel(identity, processModelId);
   }
 
+  public async getUserTasksForProcessInstance(identity: IIdentity, processInstanceId: string): Promise<DataModels.UserTasks.UserTaskList> {
+    return this._consumerApiService.getUserTasksForProcessInstance(identity, processInstanceId);
```
Comment (robpal94) [inline typescript/src/accessors/internal_accessor.ts]: See above

File: typescript/src/accessors/internal_accessor.ts [hunk 3]
```diff
@@ -216,6 +224,10 @@ export class InternalAccessor implements IConsumerApiAccessor {
     return this._consumerApiService.getManualTasksForProcessModel(identity, processModelId);
   }
 
+  public async getManualTasksForProcessInstance(identity: IIdentity, processInstanceId: string): Promise<DataModels.ManualTasks.ManualTaskList> {
+    return this._consumerApiService.getManualTasksForProcessInstance(identity, processInstanceId);
```
Comment (robpal94) [inline typescript/src/accessors/internal_accessor.ts]: See above

File: typescript/src/consumer_api_client_service.ts [hunk 4]
```diff
@@ -158,6 +158,12 @@ export class ConsumerApiClientService implements IConsumerApi {
     return this.consumerApiAccessor.getProcessModelById(identity, processModelId);
   }
 
+  public async getProcessModelByProcessInstanceId(identity: IIdentity, processInstanceId: string): Promise<DataModels.ProcessModels.ProcessModel> {
+    this._ensureIsAuthorized(identity);
+
+    return this.consumerApiAccessor.getProcessModelByProcessInstanceId(identity, processInstanceId);
```
Comment (robpal94) [inline typescript/src/consumer_api_client_service.ts]: See above

File: typescript/src/consumer_api_client_service.ts [hunk 5]
```diff
@@ -243,6 +249,12 @@ export class ConsumerApiClientService implements IConsumerApi {
     return this.consumerApiAccessor.getUserTasksForProcessModel(identity, processModelId);
   }
 
+  public async getUserTasksForProcessInstance(identity: IIdentity, processInstanceId: string): Promise<DataModels.UserTasks.UserTaskList> {
+    this._ensureIsAuthorized(identity);
+
+    return this.consumerApiAccessor.getUserTasksForProcessInstance(identity, processInstanceId);
```
Comment (robpal94) [inline typescript/src/consumer_api_client_service.ts]: See above

File: typescript/src/consumer_api_client_service.ts [hunk 6]
```diff
@@ -284,6 +296,12 @@ export class ConsumerApiClientService implements IConsumerApi {
     return this.consumerApiAccessor.getManualTasksForProcessModel(identity, processModelId);
   }
 
+  public async getManualTasksForProcessInstance(identity: IIdentity, processInstanceId: string): Promise<DataModels.ManualTasks.ManualTaskList> {
+    this._ensureIsAuthorized(identity);
+
+    return this.consumerApiAccessor.getManualTasksForProcessInstance(identity, processInstanceId);
```
Comment (robpal94) [inline typescript/src/consumer_api_client_service.ts]: See above

File: typescript/src/accessors/internal_accessor.ts (hunk 1 above)
Comment (SteffenKn) [reply typescript/src/accessors/internal_accessor.ts]: The `await` would not do anything in this case.

**Pull request merged**